common/thorhelper/thorstrand.cpp

```diff
@@ -0,0 +1,1353 @@
+
+protected:
+    Owned<IStrandJunction> input;
+    Owned<IStrandJunction> output;
+};
+
+//---------------------------------------------------------------------------------------------------------------------
+
+class CUnorderedStrandBranch : public CInterfaceOf<IStrandBranch>
+{
+public:
+    CUnorderedStrandBranch(roxiemem::IRowManager & rowManager, unsigned _numStrands, unsigned _blockSize)
+    {
+        input.setown(createStrandJunction( rowManager, 1, _numStrands, _blockSize, false));
+        output.setown(createStrandJunction( rowManager, _numStrands, 1, _blockSize, false));
+    }
+
+    virtual IStrandJunction * queryInputJunction()
+    {
+        return input;
+    }
+
+    virtual IStrandJunction * queryOutputJunction()
+    {
+        return output;
+    }
+
+protected:
+    Owned<IStrandJunction> input;
+    Owned<IStrandJunction> output;
+};
+
+//---------------------------------------------------------------------------------------------------------------------
+
+extern THORHELPER_API IStrandBranch * createStrandBranch(roxiemem::IRowManager & rowManager, unsigned numStrands, unsigned blockSize, bool isOrdered, bool isGrouped)
+{
+    //Slightly inefficient to go via a junction, but makes the testing code simpler!
+    assertex(numStrands);
+    if (numStrands == 1)
+        return new CSingleStrandBranch();
+    if (isOrdered)
+        return new COrderedStrandBranch(rowManager, numStrands, blockSize, isGrouped);
+    else if (!isGrouped)
+        return new CUnorderedStrandBranch(rowManager, numStrands, blockSize);
+    UNIMPLEMENTED;
+}
+
+//---------------------------------------------------------------------------------------------------------------------
+
+
+class BlockedRowStreamWriter : public CInterfaceOf<IRowWriterEx>
+{
+public:
+    BlockedRowStreamWriter(IRowQueue * _queue, RowBlockAllocator & _allocator) : queue(_queue), allocator(_allocator)
+    {
+        curBlock = NULL;
+    }
+
+    virtual void putRow(const void *row)
+    {
+        if (!curBlock)
+            curBlock = allocator.newBlock();
+        if (curBlock->addRowNowFull(row))
+        {
+            if (!queue->enqueue(curBlock))
+                curBlock->releaseBlock();
+            curBlock = NULL;
+        }
+    }
+
+    virtual void flush()
+    {
+        if (curBlock)
+        {
+            if (!queue->enqueue(curBlock))
+                curBlock->releaseBlock();
+            curBlock = NULL;
+        }
+    }
+
+    virtual void noteStopped()
+    {
+        flush();
+        queue->noteWriterStopped();
+    }
+
+protected:
+    IRowQueue * queue;
+    RowBlockAllocator & allocator;
+    RoxieRowBlock * curBlock;
+};
+
+class UnorderedManyToOneRowStream : public CInterfaceOf<IManyToOneRowSteam>
+{
+public:
+    UnorderedManyToOneRowStream(roxiemem::IRowManager & _rowManager, unsigned _numStrands, unsigned blockSize) : numStrands(_numStrands), allocator(_rowManager, blockSize)
+    {
+        const unsigned maxQueued = 60 - numStrands;
```

Comment:
It is completely arbitrary (due to some old code).  It should be changed to something more sensible.